### PR TITLE
readme updates for PR #15

### PR DESCRIPTION
**Pull request by dcgleim**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -128,12 +128,29 @@ depend on for other nodes.
...
+When `group` is given a factory function, it allows you to generate a group while the FetchTree is being processed.
+`factory` is given any `depends` parameters (much like `selector` and `loader`), which is useful for creating a group that
+depends on component properties, or loaded values of other resources.
+
+```js
+const todoLoader = loader(/* assume it needs an ID */)
+
+const resources = group({
+    allTodos: depends(
+        [fromProps('ids')],
```
Comment (AsaAyers) [inline README.md]: While `fromProps` works, I think it's better to depend on a `virtual` resource instead. I think gathering the dependent props at the top of the group (`ids: virtual(fromProps('ids'))`) helps document what is required.

We also talked about adding validation to `fetchTree`, I think that would be easier to do if `fromProps` isn't buried inside a `depends()`.

File: README.md [hunk 2]
```diff
@@ -172,62 +189,53 @@ const resources = group({
...
+```
 const resources = group({
-    allTodos: depends(
-        [depends.prop('todoIds')],
-        lazy((ids) => group(ids.map(
-            // yup, this is as awkward as it looks :(
-            id => depends([depends.value(id)], todoLoader)
-        )))
+    todos: depends(
+        fromProps('category.todoId'),
```
Comment (AsaAyers) [inline README.md]: This needs to be an array, and like above I think `todoId: virtual(fromProps('category.todoId'))` would be better.

File: README.md (hunk 1 above)
Comment (dcgleim) [inline README.md]: Hm, those are good points.  I was trying to focus the example to just what I was documenting here.  If you think it's such a bead idea we should discourage it by not using it in examples, should we just drop the ability to specify anything beside sibling resources altogether?

Comment (dcgleim) [inline README.md]: Oh right.  Or, as I'd look at it, you wouldn't need every component to build it themselves, as long as they accept the awkward property `currentUser: { currentUser: { ... } } }`. And now I also remember our conversation about a node that instead maps resources to properties passed to the component.

**Pull request merged**